app/models/nic/managed.rb

```diff
@@ -15,7 +15,7 @@ class Managed < Interface
     # this ensures our orchestration works on both a host and a managed interface
     delegate :progress_report_id, :capabilities, :compute_resource,
              :operatingsystem, :configTemplate, :jumpstart?, :build, :build?, :os, :arch,
-             :image_build?, :pxe_build?, :pxe_build?, :token, :to => :host
+             :image_build?, :pxe_build?, :pxe_build?, :token, :to_ip_address, :model, :to => :host
```

Comment:
Inside the `boot_server` method which determines the nextServer value.  It resolves the TFTP smart proxy's hostname using the host's resolver config.